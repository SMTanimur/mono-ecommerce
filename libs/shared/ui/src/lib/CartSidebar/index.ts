import { CartSidebar, CartSidebarProps } from './CartSidebar';

export { CartSidebar };
export type { CartSidebarProps };
