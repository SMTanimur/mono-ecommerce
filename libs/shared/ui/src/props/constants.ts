import { ButtonProps } from '@ecommerce/shared/ui/Button';
import {CartItemProps} from "@ecommerce/shared/ui/CartItem";
import {ProductSidebarProps} from "@ecommerce/shared/ui/ProductSidebar";
import {CartSidebarProps} from "@ecommerce/shared/ui/CartSidebar";
import {CategoryCardProps} from "@ecommerce/shared/ui/CategoryCard";
import {IconProps} from "@ecommerce/shared/ui/Icon";
import {ItemCardProps} from "@ecommerce/shared/ui/ItemCard";
import {NavButtonProps} from "@ecommerce/shared/ui/NavButton";
import {ProductGalleryProps} from "@ecommerce/shared/ui/ProductGallery";

import cat from './cat.jpg';
import cat2 from './cat2.jpeg';

console.log(cat2)

export const PRODUCT: ProductSidebarProps['product'] = {
  slug: 'some-name',
  price: 65,
  name: 'Some name',
  images: {
    data: [
      {
        attributes: {
          url: cat,
        },
      },
      {
        attributes: {
          url: cat2,
        },
      },
    ],
  },
  smallDescription: 'Small Description',
  description: 'Description',
  information: [
    {
      title: 'Title1',
      content: 'content1',
    },
  ],
};

export const ITEM: ItemCardProps['item'] = {
  slug: 'cat-cat',
  price: 65,
  name: 'cat cat',
  images: {
    data: [
      {
        attributes: {
          url: cat,
        },
      },
      {
        attributes: {
          url: cat2,
        },
      },
    ],
  },
  smallDescription: 'smallDescription',
  description: 'description',
};

export const BUTTON_DEFAULT_PROPS: ButtonProps = {
  children: 'hello',
  tag: 'button',
  size: 'base',
  type: 'primary',
  round: 'full',
};

export const CART_ITEM_DEFAULT_PROPS: CartItemProps = {
  inCartAmount: 1,
  item: PRODUCT,
  handleCartMinusClick: () => {
    console.log('cart minus');
  },
  handleCartPlusClick: () => {
    console.log('cart plus');
  },
  handleRemoveFromCart: () => {
    console.log('cart remove');
  },
  toggleFavorite: () => {
    console.log('favorite');
  },
};

export const CART_SIDEBAR_DEFAULT_PROPS: CartSidebarProps = {
  cartTotal: 100,
  isUser: true,
  handleCheckoutClick: async () => {
    console.log('checkout');
  },
  isLoading: false,
};

export const CATEGORY_CARD_DEFAULT_PROPS: CategoryCardProps = {
  category: {
    smallDescription:
      'smalll des',
    image: {
      data: {
        attributes: {
          url: cat,
        },
      },
    },
    slug: '1',
  },
};
export const ICON_DEFAULT_PROPS: IconProps = {
  icon: 'cart',
};

export const ITEM_CARD_DEFAULT_PROPS: ItemCardProps = {
  item: ITEM,
  size: 'base',
  handleCartClick: () => console.log('cart clicked'),
  handleAmountCartClick: () => console.log('amount cart clicked'),
  handleFavoriteClick: () => console.log('favorite clicked'),
};
export const NAV_BUTTON_DEFAULT_PROPS: NavButtonProps = {
  type: 'heart',
};
export const PRODUCT_GALLERY_DEFAULT_PROPS: ProductGalleryProps = {
  images: [
    {
      attributes: {
        url: cat,
      },
    },
    {
      attributes: {
        url: cat2,
      },
    },
  ],
};
