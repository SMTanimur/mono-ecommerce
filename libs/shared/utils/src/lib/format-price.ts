export const formatPrice = (price: number) =>
  `${Math.max(price, 0).toFixed(2)}$`;
