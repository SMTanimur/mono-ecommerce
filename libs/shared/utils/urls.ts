export const getApiUrl = (path = ''): string =>
  `${process.env['NX_STRAPI_IMG_URL'] ?? ''}${path}`;

export const getImageUrl = (path = ''): string =>
  `${process.env['NX_STRAPI_URL'] ?? 'http://localhost:4200'}${path}`;

export const createItemHref = (id: string): string => 
    `/i/${id}`;
