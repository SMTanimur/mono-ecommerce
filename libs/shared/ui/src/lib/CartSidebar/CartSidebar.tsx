import clsx from 'clsx';
import { formatPrice } from '@ecommerce/shared/utils/format-price';
import { Checkout } from './LocalComponents/Checkout';

export interface CartSidebarProps {
  cartTotal: number;
  isUser: boolean;
  handleCheckoutClick: () => Promise<void>;
  isLoading?: boolean;
}

export const CartSidebar: React.FC<CartSidebarProps> = (props) => {
  const { cartTotal, isUser, handleCheckoutClick, isLoading } = props;

  return (
    <aside
      className={clsx(
        'w-full',
        'flex flex-col',
        'md:(w-1/3 top-12 sticky h-full)'
      )}
    >
      <h2
        className={clsx(
          'mb-2',
          'flex justify-between items-center flex-wrap gap-1'
        )}
      >
        <span className="font-semibold text-base">Total</span>
        <span className="font-bold text-xl">{formatPrice(cartTotal)}</span>
      </h2>
      <small className="mb-8 inline-block">Excluding delivery</small>
      <Checkout
        isLoading={isLoading}
        isUser={isUser}
        handleCheckoutClick={handleCheckoutClick}
      />
    </aside>
  );
};
