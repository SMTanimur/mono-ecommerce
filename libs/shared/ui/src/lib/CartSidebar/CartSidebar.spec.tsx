import '@testing-library/jest-dom';

import { RenderResult, render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';

import { CART_SIDEBAR_DEFAULT_PROPS } from '@ecommerce/shared/props/constants';
import { formatPrice } from '@ecommerce/shared/utils/format-price';

import { CartSidebar } from './CartSidebar';

describe('CartSidebar', () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(<CartSidebar {...CART_SIDEBAR_DEFAULT_PROPS} />);
  });

  it('should render successfully', () => {
    const { baseElement } = renderResult;

    const tree = create(
      <CartSidebar {...CART_SIDEBAR_DEFAULT_PROPS} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();

    expect(baseElement).toBeTruthy();
  });

  it('should have proper price', () => {
    const { cartTotal } = CART_SIDEBAR_DEFAULT_PROPS;

    expect(screen.getByText(formatPrice(cartTotal))).toBeInTheDocument();
  });
});
