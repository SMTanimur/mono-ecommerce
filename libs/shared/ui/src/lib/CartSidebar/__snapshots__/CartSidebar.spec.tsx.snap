// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartSidebar should render successfully 1`] = `
<aside
  className="w-full flex flex-col md:(w-1/3 top-12 sticky h-full)"
>
  <h2
    className="mb-2 flex justify-between items-center flex-wrap gap-1"
  >
    <span
      className="font-semibold text-base"
    >
      Total
    </span>
    <span
      className="font-bold text-xl"
    >
      100.00$
    </span>
  </h2>
  <small
    className="mb-8 inline-block"
  >
    Excluding delivery
  </small>
  <button
    className="inline-block cursor-pointer transform-gpu transition-all active:scale-92 rounded-lg p-4 text-white bg-sky-600 hover:(bg-sky-700) active:(bg-sky-800)"
    onClick={[Function]}
    type="button"
  >
    <div
      className="h-12 w-full flex items-center justify-between"
    >
      <div
        className="h-full flex items-center justify-center font-bold text-lg select-none"
      >
        Checkout
      </div>
      <div
        aria-hidden="true"
        className="h-fit w-fit p-2 bg-white rounded-full"
      >
        <div
          aria-hidden="true"
          className="relative flex items-center select-none pointer-events-none h-4 w-4  sm:(h-6 w-6)"
        >
          <svg
            className="feather feather-arrow-right "
            fill="black"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
          >
            <g
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<line x1=\\"5\\" y1=\\"12\\" x2=\\"19\\" y2=\\"12\\"></line><polyline points=\\"12 5 19 12 12 19\\"></polyline>",
                }
              }
            />
          </svg>
        </div>
      </div>
    </div>
  </button>
</aside>
`;
